src/main/java/hudson/plugins/parameterizedtrigger/Plugin.java

```diff
@@ -20,6 +22,7 @@
 
 import static org.apache.commons.lang.StringUtils.isEmpty;
 
+@SuppressFBWarnings(value="NM_SAME_SIMPLE_NAME_AS_SUPERCLASS", justification="Part of public API, reviewed all uses of Plugin are fully quantified in this class")
 public class Plugin extends hudson.Plugin {
```

Comment:
Note that extending `Plugin` is deprecated. https://github.com/jenkinsci/parameterized-trigger-plugin/blob/5923a07ad166c3d77a4af0d3eab6d1a8c0305bdc/src/main/java/hudson/plugins/parameterizedtrigger/Plugin.java#L31-L32 _could_ be made into an `@Initializer`, though it could probably simply be a `static` initializer in `ResultCondition.java`.

Reply:
I understood that extending `Plugin` is deprecated. 
Can you please help me in understanding what is our concern with `Stapler.CONVERT_UTILS.register` and how would that be addressed by making it into an `@Initializer`?

I read about `CONVERT_UTILS` from its [javadocs](https://javadoc.jenkins-ci.org/component/stapler/org/kohsuke/stapler/Stapler.html) and since ResultCondition is an enum, maybe we are trying to register it so that it becomes easier for Stapler to bind the enum constants...?

Reply:
(To be clear, this is just an aside about an existing deprecated API, certainly not something that needs to be changed in this PR.)

Reply:
Thanks for the detailed info! This makes things little bit more clear to me now.